docker/build_docker.sh

```diff
@@ -356,44 +367,34 @@ function docker_build {
             ${DOCKER_SRC_DIR}/post_github_comment.py ${image_full_name} "--is_contribution"
             cat << EOF
 -------------------------
-
-Docker image [$image_full_name] has been saved as an artifact. It is available at the following link: 
-https://output.circle-artifacts.com/output/job/${CIRCLE_WORKFLOW_JOB_ID}/artifacts/0/docker_images/$IMAGENAMESAVE.gz
-
-Load it locally into docker by running:
-
-\`\`\`
-curl -L "https://output.circle-artifacts.com/output/job/${CIRCLE_WORKFLOW_JOB_ID}/artifacts/0/docker_images/$IMAGENAMESAVE.gz" | gunzip | docker load
-\`\`\`
-
+Docker image [$image_full_name] has been saved as an artifact.
 --------------------------
 EOF
         fi
     fi
 
 }
 
-# default compare circle branch against master
-DIFF_COMPARE=origin/master...${CIRCLE_BRANCH}
+# default compare branch against master
+DIFF_COMPARE=origin/master...${CI_COMMIT_SHA}
 
-if [ -z "$CIRCLE_SHA1" ]; then
-    echo "CIRCLE_SHA1 not set. Assuming local testing."
-    CIRCLE_SHA1=testing
+if [ -z "${CI_COMMIT_SHA}" ]; then
+    echo "CI_COMMIT_SHA not set. Assuming local testing."
+    CI_COMMIT_SHA=testing
     DOCKER_ORG=${DOCKER_ORG:-devtesting}
     
-    if [ -z "$CIRCLE_BRANCH" ]; then
+    if [ -z "${CI_COMMIT_REF_NAME}" ]; then
         # simply compare against origin/master
         DIFF_COMPARE=origin/master
     fi
 fi
 
-
-if [[ ! $(which pyenv) ]] && [[ -n "${CIRCLECI}" ]]; then 
-    echo "pyenv not found. setting up necessary env for pyenv on circle ci";\
+if [[ ! $(which pyenv) ]] && [[ -n "${GITLAB_CI}" ]]; then
+    echo "pyenv not found. setting up necessary env for pyenv on CI";\
     export PATH="$HOME/.pyenv/bin:$PATH"
     eval "$(pyenv init -)"
     eval "$(pyenv virtualenv-init -)"
-    pyenv shell system $(pyenv versions --bare | grep 3.10)
+    pyenv shell system "$(pyenv versions --bare | grep 3.10)"
```

Comment:
Can the magic value of `3.10` be set at the top of the file, or somewhere where it stands out?